AddressSpace/include/ArrayTools.h

```diff
@@ -0,0 +1,67 @@
+/* © Copyright CERN, 2018.  All rights not expressly granted are reserved.
+ * ArrayTools.h
+ *
+ *  Created on: Jan 15, 2018
+ *      Author: mludwig
+ *
+ *  This file is part of Quasar.
+ *
+ *  Quasar is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public Licence as published by
+ *  the Free Software Foundation, either version 3 of the Licence.
+ *
+ *  Quasar is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public Licence for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with Quasar.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef QUASAR_COMMON_SRC_ARRAYUTILS_H_
+#define QUASAR_COMMON_SRC_ARRAYUTILS_H_
+
+#include <vector>
+#include <uavariant.h>
+
+namespace AddressSpace
+{
+
+/* Note: Boolean and Byte converters are not done through overloading because in the UASDK they both resolve to the same primitive type (unsigned char). */
+
+class ArrayTools
```

Comment:
Would ArrayTools be better as a namespace? Seems like ArrayTools is not really a class: I assume we never instantiate it?

Reply:
Also, are we supporting arrays in ConfigEntries ? 

From Design.xsd:
````
    <complexType name="ConfigEntry">
    	<choice>
    		<element name="documentation" type="tns:Documentation"
    			maxOccurs="1" minOccurs="0">
    		</element>
    	</choice>
    	<attribute name="name" type="tns:VariableName" use="required"></attribute>
    	<attribute name="dataType" type="tns:ConfigEntryDataType"
    		use="required">
    	</attribute>
    	<attribute name="isKey" type="boolean"></attribute>
    	<attribute name="storedInDeviceObject" type="boolean" use="optional"></attribute>
    </complexType>
````

whereas CacheVariable looks like
```
    <complexType name="CacheVariable">
    	<all>
    		<element name="documentation" type="tns:Documentation" maxOccurs="1" minOccurs="0"></element>
    		<element name="array" type="tns:Array" maxOccurs="1" minOccurs="0"></element>
    	</all>
etc etc
```

Reply:
Thanks! You spotted a bug, or a merge error. This will get fixed soon.

Reply:
The defect has been solved, please carry on with the review/testing.

Reply:
Forgotten your previous comment about ArrayTools as a namespace. Yes, I had the same feelings. Since we both agree it's a good idea, I'll apply this (rather cosmetic) fix. Let me know the status of your review.